akka-http2-support/src/main/scala/akka/http/impl/engine/http2/PriorKnowledgeSwitch.scala

```diff
@@ -52,13 +44,14 @@ private[http] object PriorKnowledgeSwitch {
             private[this] var grabbed = ByteString.empty
             def onPush(): Unit = {
               val data = grabbed ++ grab(netIn)
-              if (data.length >= HTTP2_CONNECTION_PREFACE.length) { // We should know by now
-                if (data.startsWith(HTTP2_CONNECTION_PREFACE, 0))
+              if (data.length >= Http2Protocol.ClientConnectionPreface.length) { // We should know by now
+                if (data.startsWith(Http2Protocol.ClientConnectionPreface, 0))
                   install(http2Stack, data)
                 else
                   install(http1Stack, data)
-              } else if (data.isEmpty || data.startsWith(HTTP2_CONNECTION_PREFACE, 0)) { // Still unknown
+              } else if (data.isEmpty || Http2Protocol.ClientConnectionPreface.startsWith(data)) { // Still unknown
```

Comment:
```suggestion
              } else if (Http2Protocol.ClientConnectionPreface.startsWith(data)) { // Still unknown
```

Not needed to check for data.isEmpty here since startsWith("") returns true.